airflow/jobs/local_task_job.py

```diff
@@ -112,13 +111,6 @@ def signal_handler(signum, frame):
                                            "exceeded limit ({}s)."
                                            .format(time_since_last_heartbeat,
                                                    heartbeat_time_limit))
-
-                if time_since_last_heartbeat < self.heartrate:
```

Comment:
Ah - the the check above is done _after_ calling `self.heartbeat()` which means it may have just updated the `latest_heartbeat` -- should that check be done before calling `heartbeat()`?

Reply:
heartbeat already has this check to maintain the heart rate, so this logic is redundant for all `heartbeat()` callers. It's safe to just remove it :) The test covers this to make sure heart rate is still maintained without this check.